test/unit/io.js

```diff
@@ -933,6 +933,46 @@ describe('Input/output', function () {
     });
   });
 
+  it('TIFF setting xres and yres', function (done) {
+    const res = 300;
+    const image = sharp(fixtures.inputTiff8BitDepth)
+      .tiff({
+        xres: (res),
+        yres: (res)
+      })
+      .toFile(fixtures.outputTiff, (err, info) => {
+        if (err) throw err;
+        assert.strictEqual('tiff', info.format);
+        assert.strictEqual(image.options.tiffXres, res);
```

Comment:
Perhaps the returned `info` Object should include `xres` and `yres` properties? Or maybe the `metadata()` operation could provide this? I'd prefer to avoid testing the "private" `options` internals of a sharp instance.